runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java

```diff
@@ -326,6 +345,44 @@ private void updateCounter(SpecialWeapon specialWeapon, String name, int hit)
 		}
 	}
 
+	private void displayNotification(SpecialWeapon weapon, SpecialCounter counter)
+	{
+		if (config.receiveSpecialAttackNotification())
+		{
+			if (counter.getCount() >= getSpecialWeaponThreshold(weapon))
+			{
+				String notificationMessage = "You've reached your " + weapon.getName() + " threshold!";
+				notifier.notify(notificationMessage);
+			}
+		}
+	}
+
+	private int getSpecialWeaponThreshold(SpecialWeapon weapon)
+	{
+		int threshold = Integer.MAX_VALUE;
+
+		switch (weapon.getName())
```

Comment:
Instead of switching on the weapon name, use the weapon. This is preferred in the event an item name changes.